### ISSUE TITLE
Version VVV 3.6.0 cannot create regular file '/etc/php/7.3/fpm/php-fpm.conf': No such file or directory

### ISSUE THREAD

**Issue by DeeMacD** [Varying-Vagrant-Vagrants/VVV#2336]
## What was The Command Used To Provision
vagrant up

## What Kind of VVV Provision Was This
wordpress local site

This was an attempt to upgrade to a new version of VVV
I was having this problem already and did a box upgrade to latest version of VVV, but this did not fix the issue

## Logs/What Broke
Copying PHP configs[21m[0m
cp: cannot create regular file '/etc/php/7.3/fpm/php-fpm.conf': No such file or directory

Provisioner logs would be super helpful, copy paste the entire log including the VVV logo into a code block, or link to a gist
```
[0;38;5;2m ▷ Running the [0m[1m[0;38;5;5m'main'[21m[0;38;5;2m provisioner...[0m[0m[0m
[0m[39m[2m * Bash profile setup and directories.[21m[0m
[0m[39m[2m * Setting ownership of files in /home/vagrant to vagrant[21m[0m
[0m[39m[2m * Copying /srv/config/bash_profile                      to /home/vagrant/.bash_profile[21m[0m
[0m[39m[2m * Copying /srv/config/bash_aliases                      to /home/vagrant/.bash_aliases[21m[0m
[0m[39m[2m * Copying /srv/config/bash_aliases                      to /root/.bash_aliases[21m[0m
[0m[39m[2m * Copying /srv/config/vimrc                             to /home/vagrant/.vimrc[21m[0m
[0m[39m[2m * Copying /srv/config/subversion-servers                to /home/vagrant/.subversion/servers[21m[0m
[0m[39m[2m * Copying /srv/config/subversion-config                 to /home/vagrant/.subversion/config[21m[0m
[0m[39m[2m * Copying /srv/config/ssh_known_hosts                   to /etc/ssh/ssh_known_hosts[21m[0m
[0m[39m[2m * Copying /srv/config/sshd_config                       to /etc/ssh/sshd_config[21m[0m
[0m[39m[2m * Reloading SSH Daemon[21m[0m
[0m[39m[2m * Testing network connection to [0m[4;38;5;3mhttps://ppa.launchpad.net[0m[21m[0m
[0;38;5;2m * Successful Network connection to [0m[4;38;5;3mhttps://ppa.launchpad.net[0m[0;38;5;2m detected[0m[0m
[0m[39m[2m * Testing network connection to [0m[4;38;5;3mhttps://wordpress.org[0m[21m[0m
[0;38;5;2m * Successful Network connection to [0m[4;38;5;3mhttps://wordpress.org[0m[0;38;5;2m detected[0m[0m
[0m[39m[2m * Testing network connection to [0m[4;38;5;3mhttps://github.com[0m[21m[0m
[0;38;5;2m * Successful Network connection to [0m[4;38;5;3mhttps://github.com[0m[0;38;5;2m detected[0m[0m
[0m[39m[2m * Testing network connection to [0m[4;38;5;3mhttps://raw.githubusercontent.com[0m[21m[0m
[0;38;5;2m * Successful Network connection to [0m[4;38;5;3mhttps://raw.githubusercontent.com[0m[0;38;5;2m detected[0m[0m
[0m[39m[2m * Testing network connection to [0m[4;38;5;3mhttps://getcomposer.org[0m[21m[0m
[0;38;5;2m * Successful Network connection to [0m[4;38;5;3mhttps://getcomposer.org[0m[0;38;5;2m detected[0m[0m
[0m[39m[2m * Testing network connection to [0m[4;38;5;3mhttp://ams2.mirrors.digitalocean.com[0m[21m[0m
[0;38;5;2m * Successful Network connection to [0m[4;38;5;3mhttp://ams2.mirrors.digitalocean.com[0m[0;38;5;2m detected[0m[0m
[0;38;5;2m * Network checks succeeded[0m[0m
[0m[39m[2m * git-core/ppa already present, skipping[21m[0m
[0m[39m[2m * Setting up MySQL configuration file links...[21m[0m
[0m[39m[2m * mysql group exists[21m[0m
[0m[39m[2m * mysql user present and has uid 9001[21m[0m
[0m[39m[2m * Copying /srv/config/mysql-config/vvv-core.cnf to /etc/mysql/conf.d/vvv-core.cnf[21m[0m
[0m[39m[2m * Copying PHP configs[21m[0m
cp: cannot create regular file '/etc/php/7.3/fpm/php-fpm.conf': No such file or directory
[0;38;5;9m ! The [0m[1m[0;38;5;5m'main'[21m[0;38;5;9m provisioner ran into problems, check the full log for more details! It completed in [0m[1m[0;38;5;5m12[21m[0;38;5;9m seconds.[0m[0m
```

Also, your `config/config.yml` would be super handy -->

```yaml
---

# This file is a YAML formatted file. YAML indenting is done in spaces not
# tabs, and whitespace is significant. If you don't stick to this, it will
# fail on provision

#
# IMPORTANT, if you change this file, you have to reprovision,  no exceptions
# Do this by running either this command:
# vagrant reload --provision

# Or, if your machine is already turned on:
# vagrant provision
#

# These are your websites, and their names map on to the folders they're
# located in. See the docs for how to define these, and what all the keys
# and options are
sites:

  # latest version of WordPress, can be used for client work and testing
  # Check the readme at https://github.com/Varying-Vagrant-Vagrants/custom-site-template
  wordpress-one:
    skip_provisioning: false
    description: "A standard WP install, useful for building plugins, testing things, etc"
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    hosts:
      - one.wordpress.test
    custom:
      wpconfig_constants:
        WP_DEBUG: true
        WP_DEBUG_LOG: true
        WP_DISABLE_FATAL_ERROR_HANDLER: true # To disable in WP 5.2 the FER mode

  wordpress-two:
    skip_provisioning: false
    description: "A standard WP install, useful for building plugins, testing things, etc"
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    custom:
      # locale: it_IT
      delete_default_plugins: true
      install_plugins:
        - query-monitor
    hosts:
      - two.wordpress.test

  safaricarrentalnamibia:
    skip-provisioning: false
    description: "Safari Car Rentals Test"
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    hosts:
      - safaricarrentalnamibia.test
  

  # The wordpress-develop configuration is useful for contributing to WordPress Core.
  # It uses the built WP to serve the site
  wordpress-trunk:
    skip_provisioning: true # provisioning this one takes longer, so it's disabled by default
    description: "An svn based WP Core trunk dev setup, useful for contributor days, Trac tickets, patches"
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template-develop.git
    hosts:
      - trunk.wordpress.test

  # The following commented out site configuration will create a environment useful
  # for contributions to the WordPress meta team, .e.g WordCamps, .org, etc:
  wordpress-meta-environment:
    skip_provisioning: true # disabled by default, this takes a long time to provision
    description: "An environment useful for contributions to the WordPress meta team."
    repo: https://github.com/WordPress/meta-environment.git
    hosts:
      - wp-meta.test
    custom:
      provision_site:
        "buddypressorg.test": true
        "jobs.wordpressnet.test": true
        "wordcamp.test": true
        "wordpressorg.test": true
        "wordpresstv.test": true

  # The following commented out site configuration will create a standard WordPress
  # site in www/example-site/ available at http://my-example-site.test.
  # Remember, whitespace is significant! Tabs and spaces mean different things
  #example-site:
  #  repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
  #  hosts:
  #    - my-example-site.test

# Utilities https://varyingvagrantvagrants.org/docs/en-US/utilities/
# are system level items that aren't websites, that install tools or packages
# the core utilities install tools such as phpmyadmin
utilities:
  core: # The core VVV utility
    - tls-ca # HTTPS SSL/TLS certificates
    - phpmyadmin # Web based database client
    #- memcached-admin # Object cache management
    #- opcache-status # opcache management
    #- webgrind # PHP Debugging
    #- mongodb # needed for Tideways/XHGui
    #- tideways # PHP profiling tool, also installs xhgui check https://varyingvagrantvagrants.org/docs/en-US/references/tideways-xhgui/
    #- nvm # Node Version Manager
    #- php56
    #- php70
    #- php71
    #- php72
    - php73
    #- php74

# vm_config controls how Vagrant provisions the virtual machine, and can be used to
# increase the memory given to VVV and the number of CPU cores.
# It can also be used to override the default provider being used within Vagrant.

vm_config:
  # For WP core development we recommend at least 2GB ( 2048 ),
  # If you have 4GB of RAM, lower this to 768MB or you may encounter issues
  memory: 2048
  # CPU cores:
  cores: 2

  # this tells VVV to use the prebuilt box copied from the USB drive at contributor days
  # once set to false, do not change back to true, and reprovision
  # wordcamp_contributor_day_box: false

  # Due to a limitation within Vagrant, the specified provider is only respected on a clean `vagrant up`
  # as Vagrant currently restricts you to one provider per machine
  # https://www.vagrantup.com/docs/providers/basic_usage.html#vagrant-up
  # provider: virtualbox
  # provider: hyperv
  # provider: parallels
  # provider: vmware_desktop

# General VVV options
general:
  # Backup the databases to the database/backups subfolder on halt/suspend/destroy, set to false to disable
  db_backup: true
  # Import the databases if they're missing from backups
  db_restore: true
  # set to true to use a synced shared folder for MariaDB database storage
  db_share_type: false
  # GitHub token to use from composer
  #github_token: xxxxxx

# Settings for the vagrant plugins supported by VVV
vagrant-plugins:
  disksize: 10GB # requires the disk size vagrant plugin

```

## Possible Solution

PHP 7.3 files don't seem to be included when I pull

## Steps to Reproduce (for bugs)

1. Follow steps to update vvv - minor versions and thorough updates (https://varyingvagrantvagrants.org/docs/en-US/installation/keeping-up-to-date/)
2. git pull
3.vagrant up
4.vagrant provision
(didn't work)
5.vagrant halt
6. vagrant box update
7. git pull
6. vagrant up --provision

## Your Environment

![image](https://user-images.githubusercontent.com/14183309/104857192-945e7580-597b-11eb-9cfe-a9eb6739d66b.png)

**Comment by tomjn** [comment]
> I was having this problem already and did a box upgrade to latest version of VVV, but this did not fix the issue

A box is just a starting point for a new VM, box upgrades do nothing unless you destroy everything and start again

When you say you updated VVV, what version of VVV did you update from?

**Comment by tomjn** [comment]
imo we need to wrap the config folder copies in an if check to see if that folder exists yet. We should not create the folder if it doesn't exist and let the package create it so it has the right owner and permissions

**Comment by Mte90** [comment]
can you try a provision?

**Comment by tomjn** [comment]
@Mte90 I don't think a provision will help as this is a provisioning bug. Those folders won't exist until PHP 7.3 is installed, but it can't because it fails trying to use folders that don't exist

**Comment by Mte90** [comment]
I cannot replicate it an trying to understand maybe the upgrade to the new version didn't worked?
Or maybe the config.yml doesn't have php7.3?

**Comment by tomjn** [comment]
@Mte90 we copy the configs both before and after the PHP 7.3 install. But, we cannot copy files into a folder that has not been created yet.

So it fails copying configs before, because the folder has not been created yet

If you already have PHP 7.3 provisioned then the issue does not occur.

**Comment by tomjn** [comment]
@DeeMacD this PR _might_ fix this:

https://github.com/Varying-Vagrant-Vagrants/VVV/pull/2337

But it is untested, once you've tested and confirmed it works, let us know so we can proceed. If you don't then we'll assume it did not work and wait for further information before proceeding.

If it fails, share the full output of the provision from the terminal here

**Comment by tomjn** [comment]
@DeeMacD that PR got merged,  we've had no feedback from you though, so the assumption is that you knew how to test it, checked out the branch,  re-provisioned, and  there's a "it worked!" comment you forgot to send :)

If the `develop` branch still doesn't work let us know, otherwise this is assumed fixed by #2337
